jobs: show pause reason in SHOW JOBS + UI

dt opened the pull request:
See commits. 

In action (with #78546) on the CLI:

<img width="1396" alt="Screen Shot 2022-03-25 at 11 26 28 PM" src="https://user-images.githubusercontent.com/15615/160223030-c41b52d6-099d-431b-9c91-233057b0c9c8.png">

And on the DB Console:
<img width="970" alt="Screen Shot 2022-03-25 at 11 19 07 PM" src="https://user-images.githubusercontent.com/15615/160223055-6ce78fc8-4c58-4879-ac05-8bac0422b7b3.png">

dt committed:
jobs: return the actual error that requested a pause

Explain that the job was paused, why, and what to do about it.

Release note: none.

dt committed:
sql: add pause reason to SHOW JOBs

If a job is paused with reason, show that reason in the running_status field.

Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/78543)
<!-- Reviewable:end -->

dt committed:
ui/db-console/views/jobs: use WithErrorMessage if error is set

Paused jobs can be paused because of an error; if an error is set, even if the status is paused
rather than failed, show the error message.

Release note: none.

dt commented:
TFTR!

bors r+

dt commented:
a 22.1 backport of the last commit will either need to wait for a #78186 22.1 backport to land or will need a little rebase (which would then conflict a 78186 backport) so going to wait to see if 78186 is getting backported or not before proceeding with 22.1 backport of this change.

dt commented:
blathers backport 22.1
